tests/test_ops.py

```diff
@@ -71,3 +74,29 @@ def test_ast_convertion(self, equation, expected):
         result = make_ops_ast(indexify(eval(equation)), nfops)
 
         assert str(result) == expected
+
+
+class TestOPSLifting(object):
+
+    @pytest.mark.parametrize('eq, expected', [
+        ('Eq(u,3*a - 4**a)', 'void Kernel0(float * ut00)\n\
+{\n\
+  ut00[OPS_ACC0(0)] = -2.97015324253729F;\n\
+}'),
+        ('Eq(w,u + v)', 'void Kernel0(float * ut00, float * vt00, float * wt00)\n\
```

Comment:
If the `const` modifier is a must for the OPS kernel to all variables that only have read operations, then this test should be `void Kernel0(float * ut00,const  float * vt00, const float * wt00)`. 

Just asked Istvan in OPS channel to be sure, but if its necessary i think it should be added in this PR, because you could check it from the Expression if its on LHS or something like this.

Reply:
Good call. I had assumed it wouldn't make a difference but won't hurt to check

Reply:
I think it would be great if there is more tests. 

For example:
* Testing an expression in which the same `Function` is used more the one time. This is useful to check if the are no duplicates in the generated arguments. ( this should be working as i have seen in the code, but the test is important).
* Also test the generation of a kernel with more than 1 expression.

@FabioLuporini  could have more ideas here

Reply:
Considering Istvan answer, i think `const` modifier is necessary.

Reply:
added